### PR TITLE
Feature/pdct 614 handle vespa errors gracefully (Part Two)

### PR DESCRIPTION
**Pull request by olaughter**
Initial feedback for api users when running a query with an invalid query string. The query error has more details so we could send that back in the future too if we decide it is a good idea to share it.

This builds on work in the data access library: https://github.com/climatepolicyradar/data-access/pull/102 The version of the data access library needs updating here after that is deployed.

To Do
- [x] Bump DAL version (Ready, see: https://github.com/climatepolicyradar/navigator-backend/pull/189)

# Description

Please include:
- a summary of the changes
- links to any related issue/ticket
- any additional relevant motivation and context
- details of any dependency updates that are required for this change

## Type of change

Please select the option(s) below that are most relevant:

- [X] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

There are no vespa tests to build on, we need to come back and add some. It is a low risk change though as vespa is not live yet

## Reviewer Checklist

- [X] The PR represents a single feature (small driveby fixes are also ok)
- [X] The PR includes tests that are sufficient for the level of risk
- [X] The code is sufficiently commented, particularly in hard-to-understand areas
- [X] Any required documentation updates have been made
- [X] Any TODOs added are captured in future tickets
- [X] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/routers/search.py [hunk 1]
```diff
@@ -68,9 +69,14 @@ def _search_request(
...
             # TODO: we may wish to cache responses to improve pagination performance
-            data_access_search_response = _VESPA_CONNECTION.search(
-                parameters=data_access_search_params
-            )
+            try:
+                data_access_search_response = _VESPA_CONNECTION.search(
+                    parameters=data_access_search_params
+                )
+            except QueryError:
+                raise HTTPException(
```
Comment (diversemix) [inline app/api/api_v1/routers/search.py:77]: Do we want to log this here as well?

Comment (olaughter) [reply app/api/api_v1/routers/search.py:77]: Maybe, there is already some logging done as [part of the data access library](https://github.com/climatepolicyradar/data-access/blob/main/src/cpr_data_access/search_adaptors.py#L102-L107).  Do you have thoughts on wether we need anything further?

File: app/api/api_v1/routers/search.py [hunk 2]
```diff
@@ -68,9 +69,14 @@ def _search_request(
...
             data_access_search_params = create_vespa_search_params(db, search_body)
             # TODO: we may wish to cache responses to improve pagination performance
-            data_access_search_response = _VESPA_CONNECTION.search(
-                parameters=data_access_search_params
-            )
+            try:
+                data_access_search_response = _VESPA_CONNECTION.search(
+                    parameters=data_access_search_params
+                )
+            except QueryError:
```
Comment (THOR300) [inline app/api/api_v1/routers/search.py:76]: Do we not want except as e and then pass the error into the HttpException response?

Comment (olaughter) [reply app/api/api_v1/routers/search.py:76]: Valid question. We spoke about this and thought it might be a good future improvement

**Pull request merged**